control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -782,6 +791,43 @@ func (c *GatewayController) getRelatedHTTPRoutes(ctx context.Context, gateway ty
 	return list.Items, nil
 }
 
+func (c *GatewayController) getExternalFiltersForHTTPRoute(ctx context.Context, route gwv1beta1.HTTPRoute, resources *common.ResourceMap) ([]interface{}, error) {
+	var externalFilters []interface{}
+	for _, rule := range route.Spec.Rules {
+		for _, filter := range rule.Filters {
+			var externalFilter client.Object
+			//check to see if we need to grab this filter
+			if filter.ExtensionRef == nil {
+				continue
+			}
+			switch kind := filter.ExtensionRef.Kind; kind {
+			case "HTTPRouteRetryFilter":
+				externalFilter = &v1alpha1.HTTPRouteRetryFilter{}
+			case "HTTPRouteTimeoutFilter":
+				externalFilter = &v1alpha1.HTTPRouteTimeoutFilter{}
+			default:
+				//
+				continue
+			}
+
+			//get object from API
+			err := c.Client.Get(ctx, client.ObjectKey{
+				Name:      string(filter.ExtensionRef.Name),
+				Namespace: route.Namespace,
+			}, externalFilter)
+
+			if err != nil {
+				c.Log.Error(err, "externalref not found")
```

Comment:
This should probably be something lower than `Error`, just since it'll alarm users when the reality is that they just are missing a configuration object that they should be notified about on the route status.